### PR TITLE
feat(components/layout/grid): allow pass custom classNames

### PR DESCRIPTION
**Pull request by oriolpuig**
### Types of changes

- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### Description, Motivation and Context

If we use LayoutGrid or LayoutGridItem and we want customize some element style (for example, background/text color, etc, we need to create extra DOM elements to achieve that. 

Google loves ❤️  clean HTML, so if we let to our developers to add custom classNames, we are empowering them  to write more clean HTML and we keep Google happy! 😃

### CODE REVIEW

**Review by nistadev — APPROVED** [review]

**Review by andresz1 — APPROVED** [review]

File: components/layout/grid/test/index.test.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ describe(json.name, () => {
       expect(container.innerHTML).to.not.have.lengthOf(0)
     })
 
-    it('should NOT extend classNames', () => {
+    it('should allows you to add custom classNames to add styles', () => {
```
Comment (midudev) [inline components/layout/grid/test/index.test.js]: 🤣

File: components/layout/grid/src/gridItem/index.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import {getColSpanClassNamesTransform} from './settings.js'
 
 export default function LayoutGridItem({
   children,
+  className = '',
```
Comment (andresin87) [inline components/layout/grid/src/gridItem/index.js]: not needed, leave it as undefined (do not declare a default value)

**Review by andresin87 — CHANGES_REQUESTED** [review]
.

File: components/layout/grid/test/index.test.js (hunk 1 above)
Comment (andresin87) [inline components/layout/grid/test/index.test.js]: the test decription must only include what it test, not justify the purpose of its feature
```suggestion
    it('should allows you to add custom classNames', () => {
```

**Comment by andresin87** [comment]
conflicts!

**Review by turolopezsanabria — APPROVED** [review]
👍🏼 

Although this PR breaks one of our strongest agreements (do not extend classNames) I agree this time that it's actually a good addition to SUI Components and will solve some pains we have (positioning elements, coloring the layout, adding margins and padding, etc)

I must ping @beatrizip here, she opened this discussion back in August, thanks.

Important: This PR does not mean we will open the possibility to add `classNames` to other components.

**Review by andresin87 — APPROVED** [review]

**Pull request merged**